streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedStreamsTasks.java

```diff
@@ -384,6 +374,14 @@ void addToRestoring(final StreamTask task) {
         }
     }
 
+    private void removeTaskFromRestoring(final StreamTask task) {
```

Comment:
Some semi-related cleanup, to consolidate the state upkeep logic for all the various data structures in here

Reply:
Should we loop over `task.changelogPartitions()` instead? For StreamTask the `partitions()` and `changelogPartitions()` should be non-overlapping except the shared source/changelog partitions.

Reply:
We should also consider throwing exception here if a task spams multiple structs